### ISSUE TITLE
window.google_trackConversion is not a function

### ISSUE THREAD

**Issue by artpi** [Automattic/wp-calypso#7357]
This is error from found through Calypso remote Error logging. We have approx. 2000 of these errors per 24h

### Error message
`window.google_trackConversion is not a function`

### Paths

- /start/design-type
- /start/website/design-type
- /stats/insights/:siteId
- /
- /start/survey
- /start/domains
- /stats/day/:siteId
- /checkout/:siteId

### Browsers

- chrome
- android
- firefox
- safari

### Stack trace

> [{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"_retarget","args":[],"line":24,"column":18394,"context":"[\"errorThrown=true;this.initializeAll(0);ret=method.call(scope,a,b,c,d,e,f);errorThrown=false}finally{try{if(errorThrown){try{this.closeAll(0)}catch(err){}}else{this.closeAll(0)}}finally{this._isInTransaction=false}}return ret},initializeAll:function(start"},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"?","args":[],"line":24,"column":18083,"context":"[\"errorThrown=true;this.initializeAll(0);ret=method.call(scope,a,b,c,d,e,f);errorThrown=false}finally{try{if(errorThrown){try{this.closeAll(0)}catch(err){}}else{this.closeAll(0)}}finally{this._isInTransaction=false}}return ret},initializeAll:function(start"},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"?","args":[],"line":38,"column":22439,"context":"[\"yy:[\\\"godina\\\",\\\"godine\\\",\\\"godina\\\"]},correctGrammaticalCase:function(number,wordKey){return number===1?wordKey[0]:number>=2&&number<=4?wordKey[1]:wordKey[2]},translate:function(number,withoutSuffix,key){var wordKey=translator.words[key];if(key.length=="},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"done","args":[],"line":38,"column":20604,"context":"[\"yy:[\\\"godina\\\",\\\"godine\\\",\\\"godina\\\"]},correctGrammaticalCase:function(number,wordKey){return number===1?wordKey[0]:number>=2&&number<=4?wordKey[1]:wordKey[2]},translate:function(number,withoutSuffix,key){var wordKey=translator.words[key];if(key.length=="},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"?","args":[],"line":38,"column":19154,"context":"[\"yy:[\\\"godina\\\",\\\"godine\\\",\\\"godina\\\"]},correctGrammaticalCase:function(number,wordKey){return number===1?wordKey[0]:number>=2&&number<=4?wordKey[1]:wordKey[2]},translate:function(number,withoutSuffix,key){var wordKey=translator.words[key];if(key.length=="},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"?","args":[],"line":38,"column":22408,"context":"[\"yy:[\\\"godina\\\",\\\"godine\\\",\\\"godina\\\"]},correctGrammaticalCase:function(number,wordKey){return number===1?wordKey[0]:number>=2&&number<=4?wordKey[1]:wordKey[2]},translate:function(number,withoutSuffix,key){var wordKey=translator.words[key];if(key.length=="},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"?","args":[],"line":38,"column":28006,"context":"[\"yy:[\\\"godina\\\",\\\"godine\\\",\\\"godina\\\"]},correctGrammaticalCase:function(number,wordKey){return number===1?wordKey[0]:number>=2&&number<=4?wordKey[1]:wordKey[2]},translate:function(number,withoutSuffix,key){var wordKey=translator.words[key];if(key.length=="},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"?","args":[],"line":17,"column":19370,"context":"[\"},butLast:function(){return this.slice(0,-1)},isEmpty:function(){return this.size!==undefined?this.size===0:!this.some(function(){return true})},count:function(predicate,context){return ensureSize(predicate?this.toSeq().filter(predicate,context):this)},c"},{"url":null,"func":"Array.forEach","args":["native"],"line":null,"column":null},{"url":"https:\/\/wordpress.com\/calypso\/build-production.7ee8dcd0ac416fea9f5f.min.js","func":"HTMLScriptElement.handleCompletedRequest [as onreadystatechange]","args":[],"line":17,"column":19342,"context":"[\"},butLast:function(){return this.slice(0,-1)},isEmpty:function(){return this.size!==undefined?this.size===0:!this.some(function(){return true})},count:function(predicate,context){return ensureSize(predicate?this.toSeq().filter(predicate,context):this)},c"}]

### Kibana issues
`dadb2be6de0ee36a93aa5ddd82b28550-logstash`

**Comment by klimeryk** [comment]
This is probably the same issue as https://github.com/Automattic/wp-calypso/issues/3484 (cc @gziolo).

**Comment by gziolo** [comment]
We call `google_trackConversion` only in one place. I guess it's super easy to fix by adding a guard condition :)

**Comment by gziolo** [comment]
@klimeryk thanks for ping :)

**Comment by artpi** [comment]
I am not 100% sure that errors are originating from our code.
It MAY be olark :)

**Comment by gziolo** [comment]
I'm sure it is :) See:

https://github.com/Automattic/wp-calypso/blob/1c61e231cdbbcf35a68155631fb8444726b568e5/client/lib/analytics/ad-tracking.js#L127

**Comment by artpi** [comment]
Well, that code does not make me confident `google_trackConversion` is initialized.
It may be blocked by some userscript.

**Comment by bisko** [comment]
I guess that privacy extensions like Privacy Badger may be blocking the loading of the tracking code as a whole. 

There was a problem with the retargeting script before, where it failed with the same error when the Signup process was resumed (page reload or return to a partially completed signup), but this seems to be something else, since the issue was fixed in 45efad2dc7cc09098c66ab7c97109eaac1717044 .

Now that I look at the code again in this context, it is possible that we had a partial fix (checking if `window. google_trackConversion` exists before calling it), but was removed, as we fixed the call-before-initialization problem. 😞 

cc @mattm
